launchpad-guaranteed-tickets-v2/src/lib.rs

```diff
@@ -95,6 +95,13 @@ pub trait LaunchpadGuaranteedTickets:
         self.deposit_launchpad_tokens(total_tickets);
     }
 
+    #[endpoint(refundUserTickets)]
+    fn refund_user_tickets(&self, users_list: MultiValueEncoded<ManagedAddress>) {
+        let users_vec = users_list.to_vec();
+        self.add_users_to_blacklist(&users_vec);
+        self.clear_users_with_guaranteed_ticket_after_blacklist(&users_vec);
+    }
```

Comment:
Either simply call `add_users_to_blacklist_endpoint`, or, if the event is _that_ important, make a common function `blacklist_users` where you do the common logic and just fire the event in `add_users_to_blacklist_endpoint`

Reply:
We need the function to be without an event, like the blacklist method. I understand that it could be easier and more logical from the coding point of view to make a common function, but in this case, I think it is ok to actually have these lines separately, without what is out of scope.

Reply:
+1